### PR TITLE
OAM data documentation; move more content into data/; move palettes into gfx/; more code+script constants; consistent map naming

### PR DESCRIPTION
**Pull request by roukaour**
Also merges:

- #452 Menu work
- #457 sort the symfile by symbol type
- #459 Consistent unused content references, and predef and special label names
- #460 Replace hardcoded banks & addresses in non-mobile code

### CODE REVIEW

**Comment by roukaour** [comment]
I'm done with this PR.

**Comment by roukaour** [comment]
Okay, I'm actually done with this PR.

**Comment by roukaour** [comment]
Next change will be documenting battle anim constants, but that'll take time and be a separate PR, so merge this whenever.

**Comment by yenatch** [comment]
please put map2sym.py in another pr and rename includes.asm to constants.asm

**Comment by xCrystal** [comment]
I don't mind text/ being moved into data/, but I think all the text should remain inside a unique text/ folder.  Personally, I don't mind "deep" directory trees (3? 4? levels) nearly as much as I dislike directories with loads of files and folders all over the place. It's about organization and being able to locate content as easily as possible by navegating through directories using common sense. Whether something like phone_text.asm would belong more to a data/phone/ directory than a data/text/ directory is arguable and up to preference, but I don't think this approach or scattering all text files inside an already dense directory is a good idea; I think we should be going into the opposite direction.

**Comment by roukaour** [comment]
I'm OK with relocating the text files; will wait to see what @yenatch says.

File: home/names.asm [hunk 1]
```diff
@@ -7,7 +7,7 @@ NamesPointers:: ; 33ab
 	dbw 0, PartyMonOT
 	dbw 0, OTPartyMonOT
 	dba TrainerClassNames
-	dbw $4, $4b52 ; within PackMenuGFX
+	dbw $4, PackMenuGFX + 60
```
Comment (yenatch) [inline home/names.asm]: This comment was just to indicate that it wasn't a valid address. It's not actually referencing PackMenuGFX. My guess would be it's a pointer to MoveDescriptions (0b:4b52). The `$4, $4b52` is `$4, $4000` in Gold. The address of MoveDescriptions in Gold is also at 0x4000 (6d:4000). The bank is wrong still, but 04:4000 in Gold was GetPlayerMovement, so the bank was probably wrong to begin with.

File: engine/battle/core.asm [hunk 2]
```diff
@@ -26,7 +26,7 @@ DoBattle: ; 3c000
 	and a
 	jr z, .not_linked
 
-	ld a, [hLinkPlayerNumber]
+	ld a, [hSerialConnectionStatus]
 	cp $2
```
Comment (xCrystal) [inline engine/battle/core.asm]: Some hSerialConnectionStatus constants (like USING_INTERNAL_CLOCK) are missing in code, more specifically in engine/battle/core.asm, home/init.asm, mobile/mobile_40.asm, and a few in engine/link.asm. Pointing out in case you haven't noticed, because you may already be aware.

Comment (roukaour) [reply engine/battle/core.asm]: Thanks. Even with that the link code has much left to do (e.g. the `wPlayerLinkAction` and `wOtherPlayerLinkMode` values, the many `wXXXX` labels, hardcoded sizes like `$1b9`), but I don't have the knowledge of GB serial hardware to do it.

File: engine/pokedex.asm [hunk 3]
```diff
@@ -15,6 +15,9 @@
 	const DEXSTATE_UPDATE_UNOWN_MODE
 	const DEXSTATE_EXIT
 
+POKDEX_SCX EQU 5
```
Comment (yenatch) [inline engine/pokedex.asm]: pokdex?

File: engine/types.asm [hunk 4]
```diff
@@ -91,9 +91,32 @@ GetTypeName: ; 50964
...
-	ld bc, 13
+	ld bc, MOVE_NAME_LENGTH
 	jp CopyBytes
 ; 5097b
 
 
 INCLUDE "data/types/names.asm"
+
+
+Unreferenced_GetGen1TrainerClassName: ; 50a28
```
Comment (yenatch) [inline engine/types.asm]: what? why is this here?

Comment (Rangi42) [reply engine/types.asm]: Should it go in engine/routines/unreferenced_getgen1trainerclassname.asm?

File: wram.asm [hunk 5]
```diff
@@ -1565,7 +1565,7 @@ wDaysSince:: db
 
 SECTION "WRAM 1", WRAMX
 
-wd000:: ds 1
+wDecompressBuffer:: ds 1
```
Comment (yenatch) [inline wram.asm]: should this be a union?

Comment (Rangi42) [reply wram.asm]: No, there are no uses of unidentified `wd000` left.

Comment (yenatch) [reply wram.asm]: pretty sure wDecompressBuffer is not 1 byte large

Comment (Rangi42) [reply wram.asm]: Okay, I see. You could, but there's a big union right after it at d002, and every NEXTU would need a `ds 2` at the beginning. How about a comment noting that `wDecompressBuffer` is larger than that? (Max size: `GBCOnlyGFX` takes $540 bytes uncompressed, and ~~I think `LZ_170d16` takes $c10 in mobile_5c.asm~~. Oh, mobile_5c.asm should have used `wDecompressScratch`, not `wDecompressBuffer`. So yeah, $540 bytes, and `wGBCOnlyDecompressBuffer` would be a more accurate name.)

File: charmap.asm [hunk 6]
```diff
@@ -207,17 +208,29 @@
...
+	charmap "<TTE>",     $1e ; "って"
+	charmap "<O>",       $1e ; "を "
+	charmap "<TA!>",     $22 ; "た！"
+	charmap "<KOUGEKI>", $23 ; "こうげき"
+	charmap "<WA>",      $24 ; "は　"
+	charmap "<NO>",      $25 ; "の　"
+	charmap "<ROUTE>",   $35 ; "ばん　どうろ"
+	charmap "<WATASHI>", $36 ; "わたし"
+	charmap "<KOKO_WA>", $37 ; "ここは"
+	charmap "<GA>",      $4a ; "が "
```
Comment (yenatch) [inline charmap.asm]: I have a feeling these are a `'d` situation. It's fine to leave them for now but they were probably invisible.

File: engine/link_2.asm [hunk 7]
```diff
@@ -56,7 +56,7 @@ LinkTextbox2: ; 4d35b
 
 .PlaceBorder: ; 4d37e
 	push hl
-	ld a, $76
+	ld a, "ぁ" ; $76
```
Comment (yenatch) [inline engine/link_2.asm]: this doesnt seem like a border tile

File: engine/link_trade.asm [hunk 8]
```diff
@@ -139,7 +139,7 @@ _LoadTradeScreenBorder: ; 16d696
 
 LinkComms_LoadPleaseWaitTextboxBorderGFX: ; 16d69a
 	ld de, LinkCommsBorderGFX + $30 tiles
-	ld hl, vTiles2 tile $76
+	ld hl, vTiles2 tile "ぁ"
```
Comment (yenatch) [inline engine/link_trade.asm]: this doesnt seem right either

File: charmap.asm (hunk 6 above)
Comment (Rangi42) [reply charmap.asm]: They're dummied out in the English version, but home/text.asm and a couple of control character tables still mention them, so it's either this or `JP_ROUTE_CHAR EQU $35`.

File: engine/link_2.asm (hunk 7 above)
Comment (Rangi42) [reply engine/link_2.asm]: Yeah, I undid that later.

File: charmap.asm (hunk 6 above)
Comment (yenatch) [reply charmap.asm]: What I mean is this:
```asm
    charmap "ばん　どうろ", $35
```

Comment (Rangi42) [reply charmap.asm]: ? `charmap "ばん　どうろ", $35` doesn't exist anywhere. The `"ばん　どうろ"` is just a comment explaining what the control character was for.
